macros/src/main/scala/MacroCompiler.scala

```diff
@@ -656,9 +656,9 @@ class MacroCompilerTransform extends Transform {
   def inputForm = MidForm
   def outputForm = MidForm
 
-  def execute(state: CircuitState) = getMyAnnotations(state) match {
-    case Seq(MacroCompilerAnnotation(state.circuit.main,
-    MacroCompilerAnnotation.Params(memFile, memFileFormat, libFile, hammerIR, costMetric, mode, useCompiler, forceCompile, forceSynflops))) =>
+  def execute(state: CircuitState) = state.annotations match {
+    case Seq(anno: MacroCompilerAnnotation) =>
```

Comment:
Wait, is this right? Shouldn't we like collect the `MacroCompilerAnnotation` first?

Reply:
Did `getMyAnnotations` do the collect before?

Reply:
Yeah, collect the ones that mark this transform

Reply:
Provided you meant doing:
```
def execute(state: CircuitState) = state.annotations.collect { case a: MacroCompilerAnnotation => a } match {
```

I resolved this in 446cb84